Change FWHM plot colors, add to colors.py functionality

pllim opened the pull request:
Fix #883 . What do you think? Derived loosely from https://colorbrewer2.org/#type=diverging&scheme=RdBu&n=3

I don't think this needs a change log but let me know if you want one. Thanks.

**Updated plots**

![ginga_pick_fwhm](https://user-images.githubusercontent.com/2090236/91642274-03045c00-e9f8-11ea-941d-8d399aafdfda.jpg)

![ginga_pick_radial](https://user-images.githubusercontent.com/2090236/91642276-0566b600-e9f8-11ea-8987-e61cf96ef7e3.jpg)

ejeschke commented:
@pllim, how about [this one](https://colorbrewer2.org/#type=diverging&scheme=PRGn&n=3)? It also says colorblind-friendly.

pllim commented:
@ejeschke , that works too. Can you please tell me the proper names in your color DB that correspond to what that site is showing? Thanks.

pllim commented:
I added functionality to `colors.py` to see if I can reverse-lookup the color palette you suggested, and changed the colors. I updated the screenshot above.

Turns out, I could simply look up the colors by eye at https://matplotlib.org/3.1.0/gallery/color/named_colors.html . 🤦  I kept my code addition anyway, in case it is useful.

What do you think? Please review. Thanks!

ejeschke commented:
@pllim, you can specify colors for Ginga canvas objects by HTML hash triplets... e.g. `'#af8dc3'` and `'#7fbf7b'`

Also, the area under the curve that is shaded, to show the "width" of the FWHM, I think that was matched to the line colors in X and Y if I'm not mistaken.  Has that been updated as well? Hard to tell from this particular screenshot...

pllim commented:
> you can specify colors for Ginga canvas objects by HTML hash triplets

Not sure if Matplotlib will honor that.

> the area under the curve that is shaded

Probably not updated yet. Where is that code?

pllim commented:
Re: under the curve -- Oh wait, if it is `color3`, then should be updated already?

ejeschke commented:
> Not sure if Matplotlib will honor that.

I think matplotlib also handles hex triplets.

pllim commented:
> I think matplotlib also handles hex triplets.

Yes, I noticed too late (after I did all that code change prior). I put the hex directly in the calls now. Please review. Thanks!

ejeschke commented on ginga/util/plots.py:
```diff
@@ -467,12 +467,12 @@ def plot_fwhm(self, x, y, radius, image, cutout_data=None,
             self.logger.debug("xarr=%s" % (str(xarr)))
             fwhm_x = self._plot_fwhm_axis(xarr, iqcalc, skybg,
-                                          'blue', 'blue', 'skyblue',
+                                          '#af8dc3', '#af8dc3', 'thistle',
```
Try `#7570b3`

ejeschke commented on ginga/util/plots.py:
```diff
@@ -467,12 +467,12 @@ def plot_fwhm(self, x, y, radius, image, cutout_data=None,
             self.logger.debug("yarr=%s" % (str(yarr)))
             fwhm_y = self._plot_fwhm_axis(yarr, iqcalc, skybg,
-                                          'green', 'green', 'seagreen',
+                                          '#7fbf7b', '#7fbf7b', 'aquamarine',
```
Try `#1b9e77`

These are also suggested by the color brewer as safe for color blind. I think it is bolder and easier to read. What do you think?

Once we decide on the color, we should also modify [here](https://github.com/ejeschke/ginga/blob/master/ginga/util/plots.py#L400) with the same substitutions for `blue` and `green`. This will fix a colorblind issue for the `Radial` plot as well.

ejeschke commented on ginga/colors.py:
```diff
@@ -755,27 +757,95 @@ def recalc_color_list():
+
+    return f'#{int(rgb[0] * 255):02x}{int(rgb[1] * 255):02x}{int(rgb[2] * 255):02x}'
+
+
```
@pllim, do you think the reverse color lookup is necessary?  I feel like since we can specify hex triplets to both ginga and matplotlib there is no real reason for it.  The color list in `colors` is obviously a small subset of the 24M possible colors.  It is not like we need to find the closest, non-perfect match when you can specify an exact triplet.  If it is easier to read the code with names, can do something like
```python
green, purple = '#1b9e77', '#7570b3'
```
inside a specific module (e.g. `ginga.util.plots`) and then use the variables in the code.

I don't see much of a use case for it, but if you really like it I'm ok to leave it in...

pllim replied:
Heh heh. I guess not. It was a nice programming exercise regardless... 😬

pllim commented:
@ejeschke , I think I have addressed all your comments. I have also updated the screenshots above. And as a bonus, I threw in API doc for `colors`: https://ginga--887.org.readthedocs.build/en/887/ref_api.html#module-ginga.colors

ejeschke commented:
Thanks, @pllim! Looks good! FFTM.
